### PR TITLE
Add onboarding requests

### PR DESCRIPTION
**Pull request by santi-andracchi**
Logic is added to perform signup and singin requests. 🏛️ 

The following modifications have also been made

- The dependencies that were added using pod file are eliminated 🛑 . 
- Now only SPM is used to handle dependencies. 👋 

- Class comments are deleted, indentation is fixed. 🗒️ 

- The project is updated to the most recent version of [ios-base](https://github.com/rootstrap/ios-base) (only files that are taken into account).

### CODE REVIEW

File: blackmarket/Managers/SessionManager.swift [hunk 1]
```diff
@@ -1,41 +1,61 @@
 import UIKit
+import Combine
+import SwiftUI
```
Comment (glm4) [inline blackmarket/Managers/SessionManager.swift]: do we need SwiftUI here?

File: blackmarket/Resources/Localizable.strings [hunk 2]
```diff
@@ -32,6 +32,8 @@
 "signup_full_name_placeholder" = "Type your full name";
 "signup_confirm_password_placeholder" = "Confirm Password";
 "signup_button_title" = "Sign up";
+"signup_policy_label" = "By signing up, you accept the Data Policy and the Cookies Policy.";
```
Comment (glm4) [inline blackmarket/Resources/Localizable.strings]: I don't think we need cookies here :)

**Review by glm4 — APPROVED** [review]

File: blackmarket/Common/Models/Session.swift [hunk 3]
```diff
@@ -1,19 +1,32 @@
...
-struct Session: Codable {
-  var uid: String?
-  var client: String?
-  var accessToken: String?
-  var expiry: Date?
+class Session: NSObject, Codable {
+  @objc dynamic var uid: String?
+  @objc dynamic var client: String?
+  @objc dynamic var accessToken: String?
+  @objc dynamic var expiry: Date?
```
Comment (kstoletniy) [inline blackmarket/Common/Models/Session.swift]: what is the motivation to use use NSObject and @objc dynamic here?

File: blackmarket/Extensions/ViewControllerExtension.swift [hunk 4]
```diff
@@ -27,9 +27,9 @@ extension UIViewController {
     }
     
     guard let viewController =
-      storyboard?.instantiateViewController(withIdentifier: identifier) else {
-        assert(false, "No view controller found with that identifier")
-        return
+            storyboard?.instantiateViewController(withIdentifier: identifier) else {
+      assert(false, "No view controller found with that identifier")
```
Comment (kstoletniy) [inline blackmarket/Extensions/ViewControllerExtension.swift]: are you using Storyboard actually in the project? do you consider to start the training with programmatic views (if you're planning to start with UIKit) or directly SwiftUI? IMO Storyboard will be not longer used in futures projects anymore

File: blackmarket/Common/Models/Session.swift (hunk 3 above)
Comment (santi-andracchi) [reply blackmarket/Common/Models/Session.swift]: I brought it without thinking about it from the base project, I don't need them so I took them out

File: blackmarket/Extensions/ViewControllerExtension.swift (hunk 4 above)
Comment (santi-andracchi) [reply blackmarket/Extensions/ViewControllerExtension.swift]: I am using UIKit and the idea is to train with SwiftUI afterwards, I brought the change of the ios base project, I don't use this. So I took it out. Thanks! Maybe we can get it out of the ios base project as well.

**Pull request merged**